guides/ci/travis.md

```diff
@@ -0,0 +1,358 @@
+
+And set the environment variable:
+
+```shell-session
+$ export DATABASE_URL="mysql://root@localhost/test"
+```
+
+**Running the `specs` locally** should output 3 examples successfully. But, if we push the changes then Travis CI will report the following error: `Unknown database 'test' (Exception)`
+So, we need to configure Travis CI to use a MySQL service **and also setup the database**:
+
+>It's really **important** to notice that the lines we are adding to `.travis.yml` will depend exclusively on the development workflow we are using!
+> And remember that this is only an example using MySQL.
+
+```yaml
+# .travis.yml
+language: crystal
+crystal:
+  - latest
+
+env:
+  - DATABASE_URL="mysql://root@localhost/test"
+
+services:
+  - mysql
+
+before_install:
+  - mysql -e 'CREATE DATABASE IF NOT EXISTS test;'
+  - mysql -e 'CREATE TABLE `test`.`worlds` (`id` serial,`living_cells` int NOT NULL DEFAULT 0, PRIMARY KEY (id));'
+
+script:
+  - crystal spec
+```
+
+Pushing these changes will trigger Travis CI and the build should be successful!!
+
+We [use a `setup.sql` script](https://andidittrich.de/2017/06/travisci-setup-mysql-tablesdata-before-running-tests.html) to create a more readable `.travis.yml` file:
+
+```yaml
+# .travis.yml
+language: crystal
+crystal:
+  - latest
+
+env:
+  - DATABASE_URL="mysql://root@localhost/test"
+
+services:
+  - mysql
+
+before_install:
+  - mysql -u root --password="" < test-data/setup.sql
+
+script:
+  - crystal spec
+```
+
+and then, the file `./test-data/setup.sql` will look like this:
+
+```sql
+-- setup.sql
+CREATE DATABASE IF NOT EXISTS test;
+CREATE TABLE `test`.`worlds` (`id` serial,`living_cells` int NOT NULL DEFAULT 0, PRIMARY KEY (id));
+```
+
+## Caching
+
+If we read Travis CI job log, we will find that every time the job runs, Travis CI needs to fetch the libraries needed to run the application:
+
+```log
+Fetching https://github.com/crystal-lang/crystal-mysql.git
+Fetching https://github.com/crystal-lang/crystal-db.git
+```
+
+This takes time and on the other hand these libraries might not change as often as our application, so it looks like we may cache them and save time.
+
+Travis CI [uses caching](https://docs.travis-ci.com/user/caching/) to improve some parts of the building path. For our example. Here is the new configuration file **with cache enabled**:
+
+```yml
+# .travis.yml
+language: crystal
+crystal:
+  - latest
+
+env:
+  - DATABASE_URL="mysql://root@localhost/test"
+
+cache:
+  directories:
+    - lib
+    - $HOME/.cache/shards
+
+services:
+  - mysql
+
+before_install:
+  - mysql -u root --password="" < travisci/test.sql
+
+script:
+  - crystal spec
+```
+
+Let's push these changes. Travis CI will run, and it will install dependencies, but then it will cache the `lib/` and `$HOME/.cache/shards` folders. The following runs will use the cached dependencies.
```

Comment:
```suggestion
Let's push these changes. Travis CI will run, and it will install dependencies, but then it will cache the `$HOME/.cache/shards` folder. The following runs will use the cached dependencies.
```